lib/smbios.c

```diff
@@ -515,6 +522,47 @@ static int smbios_write_type4(ulong *current, int handle,
 	return len;
 }
 
+static int smbios_write_type7_level(ulong *current, int handle,
+				    struct smbios_ctx *ctx, int level)
+{
+	struct smbios_type7 *t;
+	int len = sizeof(struct smbios_type7);
+
+	t = map_sysmem(*current, len);
+	memset(t, 0, sizeof(struct smbios_type7));
+	fill_smbios_header(t, SMBIOS_CACHE_INFORMATION, len, handle);
+	smbios_set_eos(ctx, t->eos);
+
+	printf("=====================> write type7 via sysinfo_get_cache_info\n");
+
+	sysinfo_get_cache_info(level, t);
+
+	len = t->length + smbios_string_table_len(ctx);
+	*current += len;
+	unmap_sysmem(t);
+
+	return len;
+}
+
+static int smbios_write_type7(ulong *current, int handle,
+			      struct smbios_ctx *ctx)
+{
+	int level = 0; /* L1 by default*/
+	int i = 0;
+	int len = 0;
+
+	/* TODO: how to get the cache level without sysinfo? */
+	if (CONFIG_IS_ENABLED(SYSINFO_SMBIOS)) {
```

Comment:
So I think the new cache functionality shouldn't be bound to a sysinfo 'driver'. 
The cache stuff is angostic and should run regardless of the u-boot,sysinfo-smbios, just like the fallback code does.

Reply:
It does not depend on “u-boot,sysinfo-smbios”.
Currently the platform driver "drivers/sysinfo/smbios_plat.c" I added is independent from DT.
But it is sharing the same Kconfig `SYSINFO_SMBIOS`:
```
obj-$(CONFIG_SYSINFO_SMBIOS) += smbios.o
obj-$(CONFIG_SYSINFO_SMBIOS) += smbios_plat.o
```
So I think we still need the condition here for those platforms other than Arm.